KD_Lib/common/base_class.py

```diff
@@ -12,6 +12,23 @@ def __init__(self, teacher_model, student_model, train_loader, val_loader, optim
                  optimizer_student, loss='MSE', temp=20.0, distil_weight=0.5, device='cpu', 
                  log=False, logdir='./Experiments'):
 
+        """
+        Basic implementation of a general Knowledge Distillation framework
+
+        :param teacher_model (torch.nn.Module): Teacher model
+        :param student_model (torch.nn.Module): Student model
+        :param train_loader (torch.utils.data.DataLoader): Dataloader for training
+        :param val_loader (torch.utils.data.DataLoader): Dataloader for validation/testing
+        :param optimizer_teacher (torch.optim.*): Optimizer used for training teacher
+        :param optimizer_student (torch.optim.*): Optimizer used for training student
+        :param loss (str): Loss used for training
+        :param temp (float): Temperature parameter for distillation
+        :param distil_weight (float): Weight paramter for distillation loss
+        :param device (str): Device used for training
+        :param log (bool): True if logging required
+        :param logdir (str): Directory for storing logs
+        """
+
```

Comment:
docstrings for class are not defined in the `__init__()` method generally. They are defined under the class itself. So yours should look like <br/>

`class BaseClass:
'''
docstring 
'''
   def __init__()
`